pkg/controller/volume/persistentvolume/scheduler_assume_cache.go

```diff
@@ -158,8 +158,30 @@ func (c *assumeCache) add(obj interface{}) {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
+	if objInfo, _ := c.getObjInfo(name); objInfo != nil {
+		newVersion, err := c.getObjVersion(name, obj)
+		if err != nil {
+			glog.Errorf("add: couldn't get object version: %v", err)
+			return
+		}
+
+		storedVersion, err := c.getObjVersion(name, objInfo.latestObj)
+		if err != nil {
+			glog.Errorf("add: couldn't get stored object version: %v", err)
+			return
+		}
+
+		// Only update object if version is newer.
+		// This is so we don't override assumed objects due to informer resync.
+		if newVersion <= storedVersion {
```

Comment:
Just curious why no issue in other place? AFAIK, most of code did not handle that.

Reply:
What other places are you referring to?